documentation/SupportingDocuments/Purpose Consent Proposal comparison.md

```diff
@@ -0,0 +1,35 @@
+
+A lot of discussion around the concept of purpose and its management in CAMARA API occurred in particular in the Issue#32 (https://github.com/camaraproject/IdentityAndConsentManagement/issues/32).
+
+We have, as of now, 2 distinct proposal on the table: one initially provided by Telefonica, and then a second proposed by the Orange team. Both proposal have a lot of common & agreed on the strong requirement for purpose & consent management. But as both proposal diverge on the Open Gateway product whom frame the API commercialization, the following table sum-up main points of both proposal in a form of a FAQ.
+
+## Proposal presentation
+
+| Topic     | Telefonica proposal | Orange proposal |
+| --------- | ------------------- | --------------- |
+| What is the Open Gateway (OPGW) Product granularity? |  |  Open Gateway product matches 1-1 with a CAMARA API. For example, as of now, the CAMARA Device location API family features 2 APIs (2 yaml): `location-retrieval` and `location_verification` : we consider them as 2 separate OPGW product. The OPGW product covers completely the API (all the resources of the API). This is an important point because it means that API granularity must not be considered only from a technical aspect but also from a business aspect (which is already the case for CAMARA API). <br> The OPGW product is **not** strong-coupled with a purpose but independent of it. However in the catalog it is possible to recommend OPGW product with purpose (see next point).|
+| Do link between OPGW product and purpose is visible in Telco Catalog? |  |  In the catalog, it could be possible to browse OPGW product per category. Several categories could be created, as such 'purpose' could be defined as a category, and purpose  value could be defined as category value (in order to get for example OPGW Product related to Fraud prevention & detection) but this is only informative information.  |
+| What are country market required agreement before to launch?|   | Telco must agree on the OPGW product to be launched in the country as a minimum 'critical mass' must be available. Additionally Telco must indicate for each API scope the legal base associated with purposes for this country.|
+| What is expected during Application onboarding?|   | When a developer wishes to uses CAMARA API, developer first onboards an application. Mandatory information to onboard an application is to provide the purpose (or the list of purposes) associated with this application. |
+| What is expected during API onboarding?|   | When a developer wishes to use a CAMARA API (prequisite developer has onboarded an app - previous point), it has to contractually agrees with the Hyperscaler/agregator/telco about Open Gateway product purchase. There is not tight coupling with purpose at this stage. Of course in the catalog the Telco could recommend some API for some purpose but API (and associated Open Gateway product) and purpose are not associated.  |

```

Comment:
@bigludo7 Even if product is not tightly coupled with purpose, wouldn't the contractual agreement need to eventually tie together the App+Purpose+API-Product(with exact scopes)? Wouldn't application onboarding be a good place to bring these things & contractually agree on them?

Reply:
@shilpa-padgaonkar This is a good point to discuss. 
For me it is important to have a tight coupling between app and purpose. I guess as telco we have to make our 'homework' when a developer request to onboard an app for a given purpose.

Then when we sell API, we sell them for a given app (or a group of app) of one given developer. You have there app <-> purpose(s) and app <-> APIs. I do not have direct link between API and purpose but I'm able to know that I sold these APIs for this app which as this purpose.
Then in the `authorize` request we mandate the purpose (and I got the app indirectly via the credential)  - notably to check that we expect this purpose usage from this app (I do want to have request for a purpose not covered in the agreement). 
hope it makes sense.

Reply:
@bigludo7 
Yes exactly.  We should be able to derive that the API is used for a specific app which has agreed to a specific purpose.

What we have in mind is that at the time of app onboarding, the developer should select the purpose. He also selects the relevant API-product for the app and selects the list of relevant scopes. It is important for us that we are able to derive the legal basis at the time of onboarding the app with this combination (purpose-scopes). This combination of purpose and scopes has to lead to a single "legal basis". So when the authorize request is sent, we can check that the sent combination of purpose and scopes was exactly as was agreed during this onboarding or not. If it is not the same (if for eg. the request contains less or more scopes or if the purpose is different) we will reject the said request.

Reply:
@shilpa-padgaonkar 
Oups my bad.... should be 

> "I do **not** want to have a request for a purpose not covered in the agreement"

For us the purpose is part of the agreement for an application. This is a critical information.
Then the developer onboard for API for this application. As the on-boarded API are for **this** application they inherit for the application purpose.
As we have then the purpose (from the app), the technical scope (from the api yaml) and the country(ies) we can inform the developer about the legal base that will be applied. This legal base will be enforced in our Authorization server.

In my catalog I do not have a tight coupling between API product and purpose.
For a API usage (from an application) I have tight coupling with the application purpose and mechanistically to the legal base to apply.